Services/Language/classes/Setup/class.ilSetupLanguage.php

```diff
@@ -584,13 +584,28 @@ protected function insertLanguage($lang_key, $scope = '')
                     "(%s,%s,%s),",
                     $ilDB->quote($module, "text"),
                     $ilDB->quote($lang_key, "text"),
-                    $ilDB->quote(serialize($lang_arr), "clob"),
+                    $ilDB->quote(serialize($lang_arr), "clob")
                 );
             }
-            $ilDB->manipulate(sprintf(
-                "DELETE FROM lng_modules WHERE lang_key = %s",
-                $ilDB->quote($lang_key, "text"),
-            ));
+            $delete_query = sprintf("DELETE FROM lng_modules WHERE lang_key = %s",
+                $ilDB->quote($lang_key, "text")
+            );
+            if ($scope === "local") {
+                // delete only modules containing local changes
+                // see mantis #36972
+                $modules_to_delete = "module = '";
+                $count_modules = count($lang_array);
+                foreach ($lang_array as $module => $lang_arr) {
+                    $modules_to_delete .= $module . "'";
+                    $count_modules -= 1;
+                    if ($count_modules) {
+                        $modules_to_delete .= " OR module = '";
+                    }
+                }
+                $delete_query .= " AND (" . $modules_to_delete . ")";
```

Comment:
Please use `ilDBInterface::quote` and don't write values unescaped into the query.

Reply:
done